src/Utilities.php

```diff
@@ -135,31 +135,47 @@ private static function serializeObject(
         $maxDepth = -1,
         $depth = 0
     ) {
-        $serialized = null;
         
         if (self::serializedAlready($obj, $objectHashes)) {
-            $serialized = self::circularReferenceLabel($obj);
-        } else {
-            if ($obj instanceof \Serializable) {
-                self::markSerialized($obj, $objectHashes);
-                $serialized = $obj->serialize();
-            } else {
-                $serialized = array(
-                    'class' => get_class($obj)
-                );
-                
-                if ($obj instanceof \Iterator) {
-                    $serialized['value'] = 'non-serializable';
-                } else {
-                    $serialized['value'] = self::serializeForRollbar(
-                        $obj,
-                        $customKeys,
-                        $objectHashes,
-                        $maxDepth,
-                        $depth+1
-                    );
-                }
+            return self::circularReferenceLabel($obj);
+        }
+
+        if ($obj instanceof \Serializable) {
+            self::markSerialized($obj, $objectHashes);
+            return $obj->serialize();
+        }
+        
+        $serialized = array(
+            'class' => get_class($obj)
+        );
+        
+        if ($obj instanceof \Iterator) {
+            $serialized['value'] = 'non-serializable';
+            return $serialized;
+        }
+        
+        try {
+            if (method_exists($obj, 'toString')) {
+                $serialized['value'] = $obj->toString();
```

Comment:
The method `toString` we would _think_ would return a `string`, however that's not guaranteed: someone could do `function toString(): resource { return fopen($this->path, 'r'); }`. Recommend that the output of `$obj->toString()` pass through `serializeForRollbar` to ensure it's serialized. That may require some juggling of the checks.